sql/src/main/java/io/crate/execution/jobs/DistResultRXTask.java

```diff
@@ -183,9 +147,16 @@ private void triggerConsumer() {
         }
     }
 
-    private void mergeAndTriggerConsumer() {
+    private void merge(Map<Integer, Bucket> bucketsByIdx) {
         try {
-            mergeBuckets();
+            List<KeyIterable<Integer, Row>> buckets = new ArrayList<>(numBuckets);
+            synchronized (lock) {
+                for (Map.Entry<Integer, Bucket> entry : bucketsByIdx.entrySet()) {
+                    buckets.add(new KeyIterable<>(entry.getKey(), entry.getValue()));
+                }
+                bucketsByIdx.clear();
```

Comment:
cant we create the buckets inside the receiver and trigger the `pagingIterator.merge(buckets)` from inside?